pkg/util/cluster/cluster.go

```diff
@@ -730,6 +1019,9 @@ func (c *Cluster) fixupNSGs(ctx context.Context, vnetResourceGroup, clusterName
 }
 
 func (c *Cluster) deleteRoleAssignments(ctx context.Context, vnetResourceGroup, clusterName string) error {
+	if c.Config.UseWorkloadIdentity {
+		c.log.Print("Skipping deletion of classic role assignmnets")
```

Comment:
```suggestion
		c.log.Print("Skipping deletion of service principal role assignments")
```

See context in a different comment.